api/s2n.h

```diff
@@ -181,6 +181,8 @@ extern int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uin
 extern int s2n_connection_set_session(struct s2n_connection *conn, const uint8_t *session, size_t length);
 extern int s2n_connection_get_session(struct s2n_connection *conn, uint8_t *session, size_t max_length);
 extern ssize_t s2n_connection_get_session_length(struct s2n_connection *conn);
+extern const uint8_t *s2n_connection_get_session_id(struct s2n_connection *conn);
```

Comment:
Session id might vary in size and it can include any character, so caller needs to know the size of session id to use it.

Reply:
yeah, correct I am going to change this function to return session_id_length, that should give us an indication if we have session_id or not and we can add the function to retrieve the session_id later if needed.